### PR TITLE
[monit] Build from source and patch to use MemAvailable value if available on system

### PR DESCRIPTION
**Pull request by jleveque**
**- What I did**

Monit now reports a more accurate percentage of used system memory. Without this patch, monit doesn't seem to properly take filesystems mounted as tmpfs into consideration when calculating used system memory.

**- How I did it**

Rather than installing the latest available monit Debian package (version 5.20.0-6 for Stretch), we instead clone the source tree, patch the source so as to use the "MemAvailable" value in /proc/meminfo if it is available, build the package from source and finally install our custom-built package in the SONiC host OS.

**- How to verify it**

Verify that the percentage of used system memory displayed by monit now matches the amount returned by other utilities, such as the `virtual_memory()` function in the Python psutil module.

### CODE REVIEW

**Comment by lguohan** [comment]
looks like some build errors

**Comment by jleveque** [comment]
Fixed. Builds have all passed.

File: src/monit/Makefile [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+.SHELLFLAGS += -e
+
+MAIN_TARGET = monit_$(MONIT_VERSION)_$(CONFIGURED_ARCH).deb
+DERIVED_TARGETS = monit-dbgsym_$(MONIT_VERSION)_$(CONFIGURED_ARCH).deb
+
+$(addprefix $(DEST)/, $(MAIN_TARGET)): $(DEST)/% :
+	# Remove any stale files
+	rm -rf ./monit
+
+	# Clone isc-dhcp repo
```
Comment (lguohan) [inline src/monit/Makefile]: monit repo?

Comment (jleveque) [reply src/monit/Makefile]: Fixed. Thanks for the catch!

**Comment by jleveque** [comment]
Retest vs please

**Comment by jleveque** [comment]
Retest vsimage please

**Comment by jleveque** [comment]
Retest vsimage please

**Comment by jleveque** [comment]
Retest vsimage please

**Comment by jleveque** [comment]
Retest vsimage please

**Comment by jleveque** [comment]
This feature was merged into the upstream Monit codebase 4/2/20: https://bitbucket.org/tildeslash/monit/commits/1e420ea44a416cf23027fa850faf616d48a26cd6

It appears as though it will be available starting with version 5.27.0. Can confirm [here](https://mmonit.com/monit/changes/) once that version is released.

**Pull request merged**